### PR TITLE
feat(store-engine,store-engine-bro-fs,store-engine-dexie,store-engine-fs,store-engine-sqleet): Auto-increment primary keys

### PR DESCRIPTION
**Pull request by bennyn**
## Pull Request Checklist

- [x] My code is covered by tests
- [x] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/store-engine/src/main/engine/LocalStorageEngine.ts [hunk 1]
```diff
@@ -64,10 +65,15 @@ export class LocalStorageEngine implements CRUDEngine {
             throw new RecordAlreadyExistsError(message);
           } else {
             if (typeof record === 'string') {
-              window.localStorage.setItem(key, String(entity));
+              window.localStorage.setItem(`${internalPrimaryKey}`, String(entity));
```
Comment (lipis) [inline packages/store-engine/src/main/engine/LocalStorageEngine.ts]:
```suggestion
              window.localStorage.setItem(internalPrimaryKey, String(entity));
```

File: packages/store-engine/src/main/engine/LocalStorageEngine.ts [hunk 2]
```diff
@@ -64,10 +65,15 @@ export class LocalStorageEngine implements CRUDEngine {
...
             } else {
-              window.localStorage.setItem(key, JSON.stringify(entity));
+              window.localStorage.setItem(`${internalPrimaryKey}`, JSON.stringify(entity));
```
Comment (lipis) [inline packages/store-engine/src/main/engine/LocalStorageEngine.ts]:
```suggestion
              window.localStorage.setItem(internalPrimaryKey, JSON.stringify(entity));
```

Comment (lipis) [reply packages/store-engine/src/main/engine/LocalStorageEngine.ts]: or it needs to be a string?

File: packages/store-engine/src/main/engine/LocalStorageEngine.ts [hunk 3]
```diff
@@ -78,8 +84,8 @@ export class LocalStorageEngine implements CRUDEngine {
 
   public delete<PrimaryKey = string>(tableName: string, primaryKey: PrimaryKey): Promise<PrimaryKey> {
     return Promise.resolve().then(() => {
-      const key: string = this.createKey(tableName, primaryKey);
-      window.localStorage.removeItem(key);
+      const key = this.createKey<PrimaryKey>(tableName, primaryKey);
+      window.localStorage.removeItem(`${key}`);
```
Comment (lipis) [inline packages/store-engine/src/main/engine/LocalStorageEngine.ts]: just `key`

File: packages/store-engine/src/main/engine/LocalStorageEngine.ts (hunk 2 above)
Comment (bennyn) [reply packages/store-engine/src/main/engine/LocalStorageEngine.ts]: it needs to be a string

File: packages/store-engine/src/main/engine/LocalStorageEngine.ts (hunk 3 above)
Comment (bennyn) [reply packages/store-engine/src/main/engine/LocalStorageEngine.ts]: but it needs to be a string ;)

**Pull request merged**